packages/search-ui/src/Events.ts

```diff
@@ -43,25 +50,25 @@ function wireUpEventHandler(
 
 type EventOptions = {
   apiConnector?: APIConnector;
-  onSearch?: (searchQuery: SearchQuery) => void;
-  onAutocomplete?: (query: AutocompleteQuery) => void;
-  onResultClick?: (result: SearchResult) => void;
-  onAutocompleteResultClick?: (result: AutocompletedResult) => void;
+  onSearch?: onSearchHook;
+  onAutocomplete?: onAutocompleteHook;
+  onResultClick?: onResultClickHook;
+  onAutocompleteResultClick?: onAutocompleteResultClickHook;
 };
 
 class Events {
   // eslint-disable-next-line  @typescript-eslint/no-explicit-any
   public search: (
-    searchQuery: SearchQuery,
+    query: SearchQuery,
     queryConfig: QueryConfig
-  ) => Promise<any>;
+  ) => Promise<ResponseState>;
   // eslint-disable-next-line  @typescript-eslint/no-explicit-any
   public autocomplete: (
-    query: { searchTerm: string },
-    queryConfig: AutocompleteQuery
-  ) => Promise<any>;
-  public resultClick: () => void;
-  public autocompleteResultClick: () => void;
+    query: AutocompleteSearchQuery,
+    queryConfig: AutocompleteQueryConfig
+  ) => Promise<AutocompleteResponseState>;
+  public resultClick: (resultParams: any) => void;
```

Comment:
types need addressing here but decided to ignore this for now until this PR has been merged.